src/System.Net.Requests/src/System/Net/FtpWebRequest.cs

```diff
@@ -0,0 +1,2544 @@
+        /// <summary>
+        /// <para>Used to query for the Response of an FTP request [async version]</para>
+        /// </summary>
+        public override IAsyncResult BeginGetResponse(AsyncCallback callback, object state)
+        {
+#if FEATURE_FTP_LOGGING
+            if(Logging.On)Logging.Enter(Logging.Web, this, "BeginGetResponse", "");
+            if(Logging.On)Logging.PrintInfo(Logging.Web, this, "BeginGetResponse", SR.GetString(SR.net_log_method_equal, m_MethodInfo.Method));
+#endif
+            if (GlobalLog.IsEnabled)
+            {
+                GlobalLog.Enter("FtpWebRequest#" + ValidationHelper.HashString(this) + "::BeginGetResponse");
+            }
+
+            ContextAwareResult asyncResult;
+#if FEATURE_FTP_TRACING
+            bool success = true;
+#endif
+                        
+            try {
+                if (m_FtpWebResponse != null)
+                {
+                    asyncResult = new ContextAwareResult(this, state, callback);
+                    asyncResult.InvokeCallback(m_FtpWebResponse);
+                    return asyncResult;
+                }
+
+                if (m_GetResponseStarted) {
+                    throw new InvalidOperationException(SR.net_repcall);
+                }
+
+                m_GetResponseStarted = true;
+                CheckError();
+
+#if FTP_FEATURE_HTTP_PROXY
+                if (ServicePoint.InternalProxyServicePoint)
+                {
+                    HttpWebRequest httpWebRequest = GetHttpWebRequest();
+#if FEATURE_FTP_LOGGING
+                    if (Logging.On) Logging.Associate(Logging.Web, this, httpWebRequest);
+#endif
+                    asyncResult = (ContextAwareResult)httpWebRequest.BeginGetResponse(callback, state);
+                }
+                else
+#endif
+                {
+                    RequestStage prev = FinishRequestStage(RequestStage.RequestStarted);
+                    asyncResult = new ContextAwareResult(true, true, this, state, callback);
+                    m_ReadAsyncResult = asyncResult;
+
+                    if (prev >= RequestStage.RequestStarted)
+                    {
+                        // To make sure the context is flowed
+                        asyncResult.StartPostingAsyncOp();
+                        asyncResult.FinishPostingAsyncOp();
+
+                        if (prev >= RequestStage.ReadReady)
+                            asyncResult = null;
+                        else
+                        {
+                            lock (m_SyncObject)
+                            {
+                                if (m_RequestStage >= RequestStage.ReadReady)
+                                    asyncResult = null;;
+                            }
+                        }
+
+                        if(asyncResult == null)
+                        {
+                            // need to complete it now
```

Comment:
Nit: in all other contract ports we've capitalized the comments and ensured that the sentences ended in "."